http/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/dao/MockSamDAO.scala

```diff
@@ -385,6 +390,16 @@ class MockSamDAO extends SamDAO[IO] {
     ev: Ask[IO, TraceId]
   ): IO[Option[UserSubjectId]] = IO.pure(None)
 
+  def addUserToProject(creatorEmail: WorkbenchEmail, googleProject: GoogleProject)(implicit
+    ev: Ask[IO, TraceId]
+  ): IO[Unit] = {
+    val authHeader = userEmailToAuthorization(creatorEmail)
+    val project = ProjectSamResourceId(googleProject)
+    IO {
+      projectUsers = projectUsers |+| Map(authHeader -> Set((project, SamPolicyName.User)))
```

Comment:
what the heck is the TIE fighter operator?

Reply:
TBH I copied that bit of logic from another place in MockSamDAO, so I'm not entirely sure myself

Reply:
but I think it functions as kind of a += in this case